overlord/snapstate/backend/link.go

```diff
@@ -405,3 +410,22 @@ func (b Backend) UnlinkComponent(cpi snap.ContainerPlaceInfo, snapRev snap.Revis
 
 	return nil
 }
+
+func (b Backend) KillSnapApps(snapName string, reason snap.AppKillReason, meter progress.Meter, tm timings.Measurer) error {
+	cli := client.New()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout.DefaultTimeout))
+	defer cancel()
+
+	var err error
+	var failures []client.AppFailure
+	timings.Run(tm, "kill-snap-apps", fmt.Sprintf("kill running apps for snap %s", snapName), func(timings.Measurer) {
+		failures, err = cli.AppsKill(ctx, []string{snapName}, syscall.SIGKILL, reason)
```

Comment:
This will overwrite previous error values so it might end up notifying of failures but then returning nil

Reply:
I don't understand, which previous error values?

Reply:
Oh my bad, I think I mistook this as looping somehow (in my defence, it was late :p). Ignore me, of course this looks fine